File: core/commands/registry/irules/udp__mss.py
# Enriched from F5 iRules reference documentation.
"""UDP::mss -- Returns the on-wire Maximum Segment Size (MSS) for a UDP connection."""

from __future__ import annotations

from ....compiler.side_effects import ConnectionSide, SideEffect, SideEffectTarget
from .._base import CommandDef
from ..models import CommandSpec, FormKind, FormSpec, HoverSnippet, ValidationSpec
from ..namespace_models import EventRequires
from ..signatures import Arity
from ._base import _IRULES_ONLY, register

_SOURCE = "https://clouddocs.f5.com/api/irules/UDP__mss.html"


@register
class UdpMssCommand(CommandDef):
    name = "UDP::mss"

    @classmethod
    def spec(cls) -> CommandSpec:
        return CommandSpec(
            name="UDP::mss",
            dialects=_IRULES_ONLY,
            hover=HoverSnippet(
                summary="Returns the on-wire Maximum Segment Size (MSS) for a UDP connection.",
                synopsis=("UDP::mss",),
                snippet="Returns the on-wire Maximum Segment Size (MSS) for a UDP connection.",
                source=_SOURCE,
                examples=(
                    "when CLIENT_ACCEPTED {\n"
                    "  if { [UDP::mss] < 1000 } {\n"
                    "    pool small_req_pool\n"
                    "  } else {\n"
                    "    pool large_req_pool\n"
                    "  }\n"
                    "}"
                ),
                return_value="Returns the on-wire Maximum Segment Size (MSS) for a UDP connection",
            ),
            forms=(
                FormSpec(
                    kind=FormKind.DEFAULT,
                    synopsis="UDP::mss",
                ),
            ),
            validation=ValidationSpec(
                arity=Arity(),
            ),
            event_requires=EventRequires(transport="udp"),
            side_effect_hints=(
                SideEffect(
                    target=SideEffectTarget.UDP_STATE,
                    reads=True,
                    connection_side=ConnectionSide.BOTH,
                ),
            ),
        )

File: rust/tcl-registry/src/commands/irules/udp__mss.rs
// tcl-lsp — a language server and toolchain for Tcl
// Copyright (C) 2026 James Deucker (bitwisecook) <https://github.com/bitwisecook>
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//
// SPDX-License-Identifier: AGPL-3.0-or-later

//! `UDP::mss` iRules command.
use crate::prelude::*;
pub const fn spec() -> CommandSpec {
    CommandSpec {
        name: "UDP::mss",
        dialects: Some(DialectSet::IRULES),
        arity: Arity::at_least(0),
        hover: Some(HoverSnippet {
            summary: "Returns the on-wire Maximum Segment Size (MSS) for a UDP connection.",
            synopsis: &["UDP::mss"],
            snippet: "Returns the on-wire Maximum Segment Size (MSS) for a UDP connection.",
            source: "https://clouddocs.f5.com/api/irules/UDP__mss.html",
            examples: "when CLIENT_ACCEPTED {\n  if { [UDP::mss] < 1000 } {\n    pool small_req_pool\n  } else {\n    pool large_req_pool\n  }\n}",
            return_value: "Returns the on-wire Maximum Segment Size (MSS) for a UDP connection",
        }),
        event_requires: Some(EventRequires {
            client_side: false,
            server_side: false,
            transport: Some("udp"),
            profiles: &[],
            also_in: &[
                "SIP_REQUEST",
                "SIP_REQUEST_SEND",
                "SIP_RESPONSE",
                "STREAM_MATCHED",
            ],
            init_only: false,
            flow: false,
            capability: None,
        }),
        forms: &[FormSpec {
            synopsis: "UDP::mss",
            ..FormSpec::DEFAULT
        }],
        side_effects: &[SideEffect {
            target: SideEffectTarget::UdpState,
            reads: true,
            connection_side: ConnectionSide::Both,
            ..SideEffect::DEFAULT
        }],
        ..CommandSpec::DEFAULT
    }
}
